src/accountsdb/db.zig

```diff
@@ -1910,18 +1910,29 @@ pub const AccountsDB = struct {
     /// writes a batch of accounts to storage and updates the index
     pub fn putAccountSlice(
         self: *Self,
-        accounts: []Account,
-        pubkeys: []Pubkey,
+        accounts: []const Account,
+        pubkeys: []const Pubkey,
         slot: Slot,
     ) !void {
         std.debug.assert(accounts.len == pubkeys.len);
         if (accounts.len == 0) return;
 
         {
+            const accounts_duped = try self.allocator.dupe(Account, accounts);
```

Comment:
Good callout - this was the result of two changes made at different points in time:
1. I duped the accounts, without cloning each one
2. I cloned the accounts later in place.
The code will be simpler applying your suggestion + a necessary edit to the code just below which needs to be changed to copy `accounts` instead of doing `account.* = try account.clone(self.allocator)`.

Reply:
aca0727f33d64115c5fc45fc5f57723ab35970a1